### PR TITLE
add ar-15 .22 conversion kit

### PR DESCRIPTION
**Pull request by GuardianDll**
#### Summary
None
#### Purpose of change
ar-15 is funky, and multiple brands offers the kits to convert it into .22 round. why not?
#### Describe the solution
Add a conversion kit, with 10 and 28 round mags offered (stats delivered from chiappa production)

### CODE REVIEW

File: data/json/items/gunmod/conversions.json [hunk 1]
```diff
@@ -397,6 +397,35 @@
     "min_skills": [ [ "weapon", 2 ] ],
     "add_mod": [ [ "brass catcher", 1 ], [ "bayonet lug", 1 ], [ "rail", 2 ], [ "sights", 1 ], [ "underbarrel", 1 ], [ "muzzle", 1 ] ]
   },
+  {
+    "id": "retool_ar15_22",
+    "type": "GUNMOD",
+    "name": { "str": ".22 upper receiver" },
+    "description": "This is a complete upper receiver assembly containing a barrel, handguard, and bolt carrier group that, given the right knowledge, is installable upon an AR-15-style rifle's lower frame.  Swappable uppers enable compatible firearms to be retooled for different cartridges and barrel lengths, with this kit furnishing a weapon with a 16-inch barrel chambered for .22 rounds.  Specialized or modified STANAG-style magazines are required to properly feed cartridges of a differe.",
```
Comment (GuardianDll) [inline data/json/items/gunmod/conversions.json:404]:
```suggestion
    "description": "This is a complete upper receiver assembly containing a barrel, handguard, and bolt carrier group that, given the right knowledge, is installable upon an AR-15-style rifle's lower frame.  Swappable uppers enable compatible firearms to be retooled for different cartridges and barrel lengths, with this kit furnishing a weapon with a 16-inch barrel chambered for .22 rounds.",
```

**Comment by Hyperseeker** [comment]
The M4-22 [might not be the best choice](https://reddit.com/r/22lr/comments/sz64b5/chiappa_m422_gen_ii_post_mortem/) unless you're decidedly making the .22LR conversion dangerous:

> Huge wall of text so tl;dr, my Chiappa M4-22 Gen II with ~4500 rounds through it had a catastrophic failure that resulted in the upper falling to pieces in my hand during a routine cleaning.

The [PSA upper](https://palmettostatearmory.com/psa-16-22-lr-1-16-nitride-13-5-lightweight-m-lok-upper-with-bcg-ch-516446934.html) might be a better choice. [The reviews](https://www.pewpewtactical.com/psa-22lr-upper-review/) aren't exceptional, but the primary complaint is a lack of compatibility with some lowers, which the game doesn't model.

File: data/json/items/magazine/22.json [hunk 2]
```diff
@@ -42,6 +42,39 @@
     "flags": [ "SPEEDLOADER" ],
     "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "22": 19 } } ]
   },
+  {
+    "id": "22_ar15mag_10",
+    "type": "MAGAZINE",
+    "name": { "str": ".22 10-round conversion magazine" },
+    "description": "A black polymer magazine for ten of .22 rounds, for AR-platform rifles with conversion kit.",
```
Comment (DoctorBoomstick) [inline data/json/items/magazine/22.json:49]:
```suggestion
    "description": "A black, 10-round polymer magazine for AR-platform rifles converted to fire .22 Long Rifle ammunition.",
```

File: data/json/items/magazine/22.json [hunk 3]
```diff
@@ -42,6 +42,39 @@
...
+    "color": "light_gray",
+    "ammo_type": [ "22" ],
+    "flags": [ "MAG_COMPACT" ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "22": 10 } } ]
+  },
+  {
+    "id": "22_ar15mag_28",
+    "type": "MAGAZINE",
+    "name": { "str": ".22 28-round conversion magazine" },
+    "description": "A black polymer magazine for 28 of .22 rounds, for AR-platform rifles with conversion kit.",
```
Comment (DoctorBoomstick) [inline data/json/items/magazine/22.json:66]:
```suggestion
    "description": "A black, 10-round polymer magazine for AR-platform rifles converted to fire .22 Long Rifle ammunition.",
```

**Review by DoctorBoomstick — CHANGES_REQUESTED** [review]

**Comment by harakka** [comment]
> The M4-22 [might not be the best choice](https://reddit.com/r/22lr/comments/sz64b5/chiappa_m422_gen_ii_post_mortem/) unless you're decidedly making the .22LR conversion dangerous

This is not meant to be that specific item, the stats are just taken from there. It stands in for AR .22lr kits in general, just like we don't have other brand+model kits.

File: data/json/items/gunmod/conversions.json [hunk 4]
```diff
@@ -420,6 +420,8 @@
     "mod_targets": [ "modular_ar15", "modular_m4_carbine", "modular_m16a4", "modular_m16_auto_rifle", "modular_m27_assault_rifle" ],
     "install_time": "5 m",
     "dispersion_modifier": 15,
+    "overwrite_min_cycle_recoil": 83,
+    "//": "AR has too big recoil for .22 round, 83 is right on the edge of being able to use blackpowder loaded and ratshot",
```
Comment (GuardianDll) [inline data/json/items/gunmod/conversions.json:424]:
```suggestion
    "//0": "AR has too big recoil for .22 round, 83 is right on the edge of being able to use blackpowder loaded and ratshot",
```

File: data/json/items/gunmod/conversions.json [hunk 5]
```diff
@@ -397,6 +397,37 @@
...
+    "price_postapoc": "5 USD",
+    "material": [ "steel", "plastic" ],
+    "symbol": ":",
+    "color": "dark_gray",
+    "location": "bore",
+    "mod_targets": [ "modular_ar15", "modular_m4_carbine", "modular_m16a4", "modular_m16_auto_rifle", "modular_m27_assault_rifle" ],
+    "install_time": "5 m",
+    "dispersion_modifier": 15,
+    "overwrite_min_cycle_recoil": 83,
+    "//": "AR has too big recoil for .22 round, 83 is right on the edge of being able to use blackpowder loaded and ratshot",
```
Comment (GuardianDll) [inline data/json/items/gunmod/conversions.json:424]:
```suggestion
    "//0": "AR has too big recoil for .22 round, 83 is right on the edge of being able to use blackpowder loaded and ratshot",
```

File: data/json/items/magazine/22.json [hunk 6]
```diff
@@ -42,6 +42,39 @@
...
+    "color": "light_gray",
+    "ammo_type": [ "22" ],
+    "flags": [ "MAG_COMPACT" ],
+    "pocket_data": [ { "pocket_type": "MAGAZINE", "rigid": true, "ammo_restriction": { "22": 10 } } ]
+  },
+  {
+    "id": "22_ar15mag_28",
+    "type": "MAGAZINE",
+    "name": { "str": ".22 28-round conversion magazine" },
+    "description": "A black, 10-round polymer magazine for AR-platform rifles converted to fire .22 Long Rifle ammunition.",
```
Comment (Holli-Git) [inline data/json/items/magazine/22.json:66]:
```suggestion
    "description": "A black, 28-round polymer magazine for AR-platform rifles converted to fire .22 Long Rifle ammunition.",
```

File: data/json/items/gunmod/conversions.json [hunk 7]
```diff
@@ -397,6 +397,37 @@
...
+    "location": "bore",
+    "mod_targets": [ "modular_ar15", "modular_m4_carbine", "modular_m16a4", "modular_m16_auto_rifle", "modular_m27_assault_rifle" ],
+    "install_time": "5 m",
+    "dispersion_modifier": 15,
+    "overwrite_min_cycle_recoil": 83,
+    "//0": "AR has too big recoil for .22 round, 83 is right on the edge of being able to use blackpowder loaded and ratshot",
+    "ammo_modifier": [ "22" ],
+    "magazine_adaptor": [ [ "22", [ "22_ar15mag_10", "22_ar15mag_28" ] ] ],
+    "pocket_mods": [ { "pocket_type": "MAGAZINE_WELL", "item_restriction": [ "22_ar15mag_10", "22_ar15mag_28" ] } ],
+    "min_skills": [ [ "weapon", 2 ] ],
```
Comment (Holli-Git) [inline data/json/items/gunmod/conversions.json:428]:
```suggestion
    "min_skills": [ [ "weapon", 2 ] ],
    "damage_modifier": 2,
```
image
.22 LR doesn't have the special ballistic by the inch data yet, so damage modifiers need to be calculated/added manually.

File: data/json/items/gunmod/conversions.json [hunk 8]
```diff
@@ -397,6 +397,38 @@
...
+    "symbol": ":",
+    "color": "dark_gray",
+    "location": "bore",
+    "mod_targets": [ "modular_ar15", "modular_m4_carbine", "modular_m16a4", "modular_m16_auto_rifle", "modular_m27_assault_rifle" ],
+    "install_time": "5 m",
+    "dispersion_modifier": 15,
+    "overwrite_min_cycle_recoil": 83,
+    "//0": "AR has too big recoil for .22 round, 83 is right on the edge of being able to use blackpowder loaded and ratshot",
+    "ammo_modifier": [ "22" ],
+    "magazine_adaptor": [ [ "22", [ "22_ar15mag_10", "  22_ar15mag_28" ] ] ],
```
Comment (GuardianDll) [inline data/json/items/gunmod/conversions.json:426]:
```suggestion
    "magazine_adaptor": [ [ "22", [ "22_ar15mag_10", "22_ar15mag_28" ] ] ],
```

**Pull request merged**